Data race in TestAgentRestart/Restart_Both_Sides

https://github.com/pion/ice/pull/306/checks?check_run_id=1468689602#step:5:66
```
=== RUN   TestAgentRestart
=== RUN   TestAgentRestart/Restart_During_Gather
=== RUN   TestAgentRestart/Restart_When_Closed
=== RUN   TestAgentRestart/Restart_One_Side
=== RUN   TestAgentRestart/Restart_Both_Sides
==================
WARNING: DATA RACE
Read at 0x00c0004c2950 by goroutine 83:
  github.com/pion/ice/v2.(*CandidateHost).Marshal()
      <autogenerated>:1 +0x73
  github.com/pion/ice/v2.copyCandidate()
      /home/runner/work/ice/ice/transport_test.go:287 +0x42
  github.com/pion/ice/v2.gatherAndExchangeCandidates()
      /home/runner/work/ice/ice/transport_test.go:194 +0x3ea
  github.com/pion/ice/v2.TestAgentRestart.func4()
      /home/runner/work/ice/ice/agent_test.go:1452 +0x69e
  testing.tRunner()
      /opt/hostedtoolcache/go/1.15.5/x64/src/testing/testing.go:1123 +0x202

Previous write at 0x00c0004c2950 by goroutine 78:
  sync/atomic.StoreInt64()
      /opt/hostedtoolcache/go/1.15.5/x64/src/runtime/race_amd64.s:234 +0xb
  sync/atomic.StorePointer()
      /opt/hostedtoolcache/go/1.15.5/x64/src/runtime/atomic_pointer.go:51 +0x33
  github.com/pion/ice/v2.(*candidateBase).setLastSent()
      /home/runner/work/ice/ice/candidate_base.go:367 +0xab
  github.com/pion/ice/v2.(*candidateBase).seen()
      /home/runner/work/ice/ice/candidate_base.go:372 +0x97
  github.com/pion/ice/v2.(*candidateBase).writeTo()
      /home/runner/work/ice/ice/candidate_base.go:306 +0x3b3
  github.com/pion/ice/v2.(*CandidateHost).writeTo()
      <autogenerated>:1 +0x89
  github.com/pion/ice/v2.(*Agent).sendSTUN()
      /home/runner/work/ice/ice/candidatepair.go:90 +0xc1
  github.com/pion/ice/v2.(*Agent).sendBindingRequest()
      /home/runner/work/ice/ice/agent.go:953 +0x4f2
  github.com/pion/ice/v2.(*controllingSelector).PingCandidate()
      /home/runner/work/ice/ice/selection.go:159 +0x57d
  github.com/pion/ice/v2.(*Agent).pingAllCandidates()
      /home/runner/work/ice/ice/agent.go:608 +0x441
  github.com/pion/ice/v2.(*controllingSelector).ContactCandidates()
      /home/runner/work/ice/ice/selection.go:65 +0x4d2
  github.com/pion/ice/v2.(*Agent).connectivityChecks.func1.1()
      /home/runner/work/ice/ice/agent.go:504 +0x184
  github.com/pion/ice/v2.(*Agent).taskLoop()
      /home/runner/work/ice/ice/agent.go:222 +0x1a1

Goroutine 83 (running) created at:
  testing.(*T).Run()
      /opt/hostedtoolcache/go/1.15.5/x64/src/testing/testing.go:1168 +0x5bb
  github.com/pion/ice/v2.TestAgentRestart()
      /home/runner/work/ice/ice/agent_test.go:1413 +0x299
  testing.tRunner()
      /opt/hostedtoolcache/go/1.15.5/x64/src/testing/testing.go:1123 +0x202

Goroutine 78 (running) created at:
  github.com/pion/ice/v2.NewAgent()
      /home/runner/work/ice/ice/agent.go:349 +0x14b4
  github.com/pion/ice/v2.pipe()
      /home/runner/work/ice/ice/transport_test.go:240 +0x255
  github.com/pion/ice/v2.TestAgentRestart.func4()
      /home/runner/work/ice/ice/agent_test.go:1426 +0xd3
  testing.tRunner()
      /opt/hostedtoolcache/go/1.15.5/x64/src/testing/testing.go:1123 +0x202
==================
    testing.go:1038: race detected during execution of test
=== CONT  TestAgentRestart
    testing.go:1038: race detected during execution of test
--- FAIL: TestAgentRestart (1.84s)
    --- PASS: TestAgentRestart/Restart_During_Gather (0.00s)
    --- PASS: TestAgentRestart/Restart_When_Closed (0.00s)
    --- PASS: TestAgentRestart/Restart_One_Side (1.21s)
    --- FAIL: TestAgentRestart/Restart_Both_Sides (0.22s)
```

--- a/candidate_base.go
+++ b/candidate_base.go
@@ -388,7 +388,7 @@ func (c *candidateBase) context() context.Context {
 }
 
 // Marshal returns the string representation of the ICECandidate
-func (c candidateBase) Marshal() string {
+func (c *candidateBase) Marshal() string {
 	val := fmt.Sprintf("%s %d %s %d %s %d typ %s",
 		c.Foundation(),
 		c.Component(),

